packages/react-conformance/src/defaultTests.tsx

```diff
@@ -327,6 +327,129 @@ export const defaultTests: TestObject = {
     });
   },
 
+  'component-has-static-classnames-object': (componentInfo: ComponentDoc, testInfo: IsConformantOptions) => {
+    const {
+      componentPath,
+      Component,
+      wrapperComponent,
+      helperComponents = [],
+      requiredProps,
+      customMount = mount,
+    } = testInfo;
+
+    const componentName = componentInfo.displayName;
+    const componentClassName = `fui-${componentName}`;
+    let handledClassNamesObjectExport = false;
+
+    it('has static classnames exported at top-level (component-has-static-classnames-object)', () => {
+      if (testInfo.isInternal) {
+        return;
+      }
+
+      const exportName =
+        componentInfo.displayName.slice(0, 1).toLowerCase() + componentInfo.displayName.slice(1) + 'ClassNames';
+
+      try {
+        const indexFile = require(path.join(getPackagePath(componentPath), 'src', 'index'));
+        const classNamesFromFile = indexFile[exportName];
+        expect(classNamesFromFile).toBeTruthy();
+        handledClassNamesObjectExport = true;
+      } catch (e) {
+        // Need to work on this error message
+        throw new Error(
+          defaultErrorMessages['component-has-static-classnames-object-exported'](
+            testInfo,
+            e,
+            componentClassName,
+            exportName,
+          ),
+        );
+      }
+    });
+
+    it('has static classnames in correct format (component-has-static-classnames-object)', () => {
```

Comment:
This test can be dropped once we support a minimum of TS4.1 as we can re-write the `SlotClassNames<T>` type as to enforce this with TS:

```ts
export type SlotClassNames<Component, Slots> = {
  [SlotName in keyof Slots]-?: SlotName extends 'root'
    ? `fui-${string & Component}`
    : `fui-${string & Component}__${string & SlotName}`;
};
```

Reply:
@ling1726 This test validates that the slot name is in the class name. This can also be accomplished with TS which is preferrable imo but I believe it requires a newer version of TS that exceeds our minimum supported version.